core/src/lib.rs

```diff
@@ -329,6 +329,84 @@ impl From<f64> for TypedValue {
     }
 }
 
+impl From<i64> for TypedValue {
+    fn from(value: i64) -> TypedValue {
+        TypedValue::Long(value)
```

Comment:
I should add to this: because `Entid` is a type alias, not a newtype, the compiler can't help. We do have a `KnownEntid` newtype, but `Entid` still exists (for good reason), and so we have multiple distinct uses of what the compiler sees as a single type.